tools/lzld/README.md

```diff
@@ -0,0 +1,36 @@
+A `ld -lazy_framework` for macOS. Designed to build Deno.
```

Comment:
Add comment to make the purpose crystal clear to future readers

> The purpose of this ld wrapper is to improve startup time on Mac. Because Deno includes WebGPU, it needs to link to Metal. We've observed that loading Metal.framework during startup can cost as much as 8ms of startup time. This library implements `-lazy_framework` .... blah blah